File: my-folder/0007-reverse-integer/solution.java
class Solution {
    public int reverse(int x) {
        int rep = Math.abs(x);
        int res = 0;
        while (rep > 0) {
            if (res > (Integer.MAX_VALUE - rep % 10) / 10) {
                return 0;
            }
            res = res * 10 + rep % 10;
            rep /= 10;
        }
        return x < 0 ? -res : res;
    }
}

File: my-folder/0007-reverse-integer/solution.py
class Solution:
    def reverse(self, x: int) -> int:
        res = 0
        c = abs(x)
        while c > 0:
            rem = c % 10
            res = res * 10 + rem
            c //= 10
        if res > 2 ** 31 - 1 or res < -2 ** 31:
            return 0
        return res if x > 0 else -res
